packages/mobile/src/store/offline-downloads/sagas/processDownloadQueueSaga/downloadCollectionWorker.ts

```diff
@@ -0,0 +1,107 @@
+import type { ID } from '@audius/common'
+import { accountSelectors, getContext } from '@audius/common'
+import { select, call, put } from 'typed-redux-saga'
+
+import {
+  downloadCollectionCoverArt,
+  DownloadCollectionError,
+  writeCollectionJson,
+  writeFavoritesCollectionJson
+} from 'app/services/offline-downloader'
+
+import type { CollectionId } from '../../slice'
+import {
+  completeDownload,
+  downloadQueuedItem,
+  errorDownload,
+  startDownload
+} from '../../slice'
+
+const { getUserId } = accountSelectors
+
+// TODO add favorites collection task
+export function* downloadCollectionWorker(collectionId: CollectionId) {
+  yield* put(startDownload({ type: 'collection', id: collectionId }))
+
+  // "favorites" collection short circuit
+  if (typeof collectionId === 'string') {
```

Comment:
Is this condition safe? Is there a better way to check for favorites collection?

Reply:
yes great catch, should just be able to check if its equal to DOWNLOAD_REASONS_FAVORITE